### ISSUE TITLE
Why my file csv is lost while using this plugin?

### ISSUE THREAD

**Issue by rizal-mujahiddan69** [grafana/grafana-csv-datasource#247]
im using grafana in windows service with filepath ../grafanalabs/data/csv/file.csv. but after im using thats file. my file in thats path already losing?

**Comment by eab** [comment]
I'm using v0.6.6 with Grafana v10.1.1 on Ubuntu.

I'm facing the same issue.
After a reboot the CSV-files in /var/lib/grafana/csv/ are gone. It looks like the directory is deleted and recreated after a reboot.

It seems that grafana-server or the grafana-csv-datasource itself does this.

root@ubuntu-lab:/var/lib/grafana/csv# ls -ltrah
total 23M
drwxr-xr-x 20 root root 4.0K Sep 19 17:59 .
drwxr-xr-x 20 root root 4.0K Sep 19 17:59 ..
-rw-r--r--  1 root root  22M Sep 20 11:25 output.csv
-rw-r--r--  1 root root 343K Sep 20 11:25 results.csv

/etc/init.d/grafana-server restart

root@ubuntu-lab:/var/lib/grafana/csv# ls -ltrah
total 8.0K
drwx------ 2 grafana grafana 4.0K Sep 25 16:14 .
drwxr-xr-x 6 grafana grafana 4.0K Sep 25 16:14 ..

It does not happen all the times.

**Comment by zoltanbedi** [comment]
@eab is this happening when the file is not under grafana's path?

**Comment by eab** [comment]
@zoltanbedi no, when the files are in for example /random/path/csv nothing happens.

**Comment by ckholy2** [comment]
I found that it's only happened if the file's modified date is over 2days (I didn't test it in details for how much time for sure)

**Comment by luis15pt** [comment]
Can confirm im seeing the same issue

**Comment by zoltanbedi** [comment]
I suggest to try to use infinity datasource or change the way you load your csv file as we might not get to this issue any time soon.

**Comment by AgnesToulet** [comment]
Hello! The `data/csv` folder (along with `data/png` and `data/pdf`) is used to store temporarily generated files from the image renderer. This is why we delete them regularly. 

As a workaround, I suggest creating a new folder (like `csvs`). We'll prioritize this issue within our squad and get back to you.
